integration-libs/s4-service/order/components/reschedule-service-order/reschedule-service-order.component.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { RescheduleServiceOrderComponent } from './reschedule-service-order.component';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  I18nTestingModule,
+  RoutingService,
+} from '@spartacus/core';
+import { of, throwError } from 'rxjs';
+import { OrderDetailsService } from '@spartacus/order/components';
+import { FormBuilder } from '@angular/forms';
+import {
+  RescheduleServiceOrderFacade,
+  CheckoutServiceSchedulePickerService,
+  ServiceDateTime,
+} from '@spartacus/s4-service/root';
+import { UrlTestingModule } from 'projects/core/src/routing/configurable-routes/url-translation/testing/url-testing.module';
+import createSpy = jasmine.createSpy;
+
+const mockOrder = {
+  code: '0005004001',
+  entries: [
+    {
+      product: {
+        code: 'SRV_01',
+        name: 'SRV_01',
+        productTypes: 'SERVICE',
+      },
+    },
+  ],
+  serviceReschedulable: false,
+  servicedAt: '2024-08-06T11:00:00+0000',
+};
+const mockOrder2 = {
+  entries: [
+    {
+      product: {
+        code: 'SRV_01',
+        name: 'SRV_01',
+        productTypes: 'SERVICE',
+      },
+    },
+  ],
+};
+class MockOrderDetailsService {
+  getOrderDetails() {
+    return of(mockOrder);
+  }
+}
+class MockRescheduleServiceOrderFacade {
+  rescheduleService(_orderCode: string, _scheduledAt: ServiceDateTime) {}
+}
+class MockRoutingService {
+  go = createSpy().and.callThrough();
+}
+class MockGlobalMessageService implements Partial<GlobalMessageService> {
+  add = createSpy().and.callThrough();
+}
+class MockCheckoutServiceSchedulePickerService {
+  getMinDateForService = createSpy().and.returnValue(
+    of('2024-07-06T11:00:00+0000')
+  );
+  getScheduledServiceTimes = createSpy().and.returnValue(
+    of(['08:00', '14:30', '16:00'])
+  );
+  convertDateTimeToReadableString =
+    createSpy().and.returnValue('11/07/2024, 14:30');
+  getServiceDetailsFromDateTime = createSpy().and.returnValue({
+    date: '11/07/2024',
+    time: '14:30',
+  });
+  convertToDateTime = createSpy().and.returnValue('2024-06-27T14:30:00±HH:MM');
+}
+
+describe('RescheduleServiceOrderComponent', () => {
+  let component: RescheduleServiceOrderComponent;
+  let fixture: ComponentFixture<RescheduleServiceOrderComponent>;
+  let rescheduleServiceOrdeFacade: RescheduleServiceOrderFacade;
```

Comment:
Correct the spelling rescheduleServiceOrderFacade .Please add it in reference as well?

Reply:
> Please add it in reference as well
Did not understand this

Reply:
All references to rescheduleServiceOrderFacade need to be corrected. I've noticed many spelling mistakes in various places.

Reply:
Corrected all spelling mistakes I could find.